usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -6882,10 +6882,10 @@ l2arc_write_buffers(spa_t *spa, l2arc_dev_t *dev, uint64_t target_sz)
 
 	ASSERT3U(write_asize, <=, target_sz);
 	ARCSTAT_BUMP(arcstat_l2_writes_sent);
-	ARCSTAT_INCR(arcstat_l2_write_bytes, write_asize);
+	ARCSTAT_INCR(arcstat_l2_write_bytes, write_psize);
 	ARCSTAT_INCR(arcstat_l2_size, write_sz);
```

Comment:
George, this is a good suggestion, but I am concerned that this might cause some trouble for external tools that monitor ZFS ARC or convert its statistics for a nice presentation, etc.  arcstats.py comes to mind.